demos/2D-IR-Raman/noneqm-response.py

```diff
@@ -0,0 +1,251 @@
+        self.chk_last = input_param.tlast
+
+        self.fmt_bead = (
+            "{0:0" + str(int(1 + np.floor(np.log(self.nbeads) / np.log(10)))) + "d}"
+        )
+
+    def process(self):
+        # Get values of dip/pol along field_pol direction and average over beads.
+        x = []
+        op_index = 0
+        for op, field_pol in zip(self.op, self.field_pol):
+            x_single = 0
+            for bead in range(self.nbeads):
+                if op_index == 0:
+                    x_single += np.loadtxt(
+                        self.sim_name + "." + op + "_" + self.fmt_bead.format(bead)
+                    )[:, field_pol]
+                else:
+                    x_single += np.loadtxt(
+                        self.sim_name
+                        + "_noneqm."
+                        + op
+                        + "_"
+                        + self.fmt_bead.format(bead)
+                    )[:, field_pol]
+            op_index += 1
+            x_single /= self.nbeads
+            x.append(x_single)
+
+        # Compute result:
+        c = 0  # Count nonequilibrium events.
+        r_t = 0
+        i_step = self.tsteps + 1  # Number of steps for t1 and t2.
+        first = (
+            self.chk_first * self.step_traj
+            if self.chk_first is not None
+            else self.step_traj
+        )
+        last = (
+            (self.chk_last + 1) * self.step_traj
+            if self.chk_last is not None
+            else len(x[0])
+        )
+        for i in range(
+            first, last, self.step_traj
+        ):  # Start at the first available checkpoint from eq dynamics.
+            if i + 1 >= i_step:  # Check if we have enough dynamics to go backward -t1.
+                # Backward equilibrium trajectory, first operator.
+                eq = np.flip(
+                    x[0][i - i_step + 1 : i + 1]
+                )  # Equilibrium dynamics part from time t (index i) to t-t1.
+                # Difference between forward nonequilibrium trajectories, second operator.
+                neq = (
+                    x[1][2 * c * i_step : (2 * c + 1) * i_step]
+                    - x[1][(2 * c + 1) * i_step : 2 * (c + 1) * i_step]
+                )  # Neq dynamics.
+                # Compute r_t[j, k] = eq[j] * neq[k].
+                r_t += np.outer(eq, neq)
+                c += 1
+        r_t *= self.beta / (c * self.epsilon)
+        r_t = np.gradient(r_t, self.step_print * self.delta_t, axis=0, edge_order=2)
+
+        np.savetxt(
+            self.out_name + self.op[0] + "_" + self.op[1] + "_neq_2d.dat",
+            r_t,
+            fmt="%.5e",
+        )
+
+
+if __name__ == "__main__":
+    from optparse import OptionParser
+
+    parser = OptionParser(
+        usage="%prog [options] <input file>",
+        description="noneqm-response comutes response function of "
+        "2D IR-Raman spectra from equilibrium and nonequilibrium trajectories",
+    )
```

Comment:
Can you also explain here the output format, i.e. what are the rows and the columns?